bundle/edu.gemini.pot/src/main/java/edu/gemini/spModel/dataflow/GsaAspect.java

```diff
@@ -46,28 +67,50 @@ public static GsaAspect getDefaultAspect(ProgramType type) {
         return (res == null) ? DEFAULT : res;
     }
 
-    private boolean _sendToGsa;
-    private int _proprietaryMonths = -1;
-    private boolean _keepHeaderPrivate;
+    /**
+     * Looks up the GsaAspect that should be associated with the  program.  This
+     * may be explicitly assigned to the program. If not, and if the program
+     * type can be derived from the program id (which may or may not be set),
+     * use the default for the program type.  If all else fails, then return
+     * a default 0 proprietary month instance.
+     */
+    public static GsaAspect lookup(ISPProgram prog) {
+        final GsaAspect gsa = ((SPProgram) prog.getDataObject()).getGsaAspect();
+        return gsa != null ? gsa :  GsaAspect.getDefaultAspect(
+            ImOption.apply(prog.getProgramID()).flatMap(pid -> ImOption.apply(ProgramType$.MODULE$.readOrNull(pid)))
+        );
+    }
+
+    private final boolean _sendToGsa;
+    private final int _proprietaryMonths;
+    private final Visibility _headerVisibility;
 
     public GsaAspect(boolean sendToGsa, int months) {
-        this(sendToGsa, months, false);
+        this(sendToGsa, months, Visibility.PUBLIC);
     }
 
-    public GsaAspect(boolean sendToGsa, int months, boolean headerPrivate) {
+    public GsaAspect(boolean sendToGsa, int months, Visibility visibility) {
```

Comment:
I switched the boolean `headerPrivate` flag for an enum but left `sendToGsa` alone.  The `sendToGsa` flag will be removed altogether in a subsequent PR.  Now all datasets are archived so it is irrelevant.